src/main/daml/Daml/Finance/Util/Lockable.daml

```diff
@@ -43,7 +42,9 @@ releaseImpl :
   , HasToInterface t Lockable
   ) => Optional Lock -> (Optional Lock -> t) -> Release -> Update (ContractId Lockable)
 releaseImpl currentLock setLock Release{context} = do
-  assertMsg "Must be locked" $ isSome currentLock
+  case currentLock of
+    None -> abort "Must be locked"
+    Some lock -> assertMsg "Context must be recognized" $ Set.member context lock.context
```

Comment:
How about

```suggestion
    Some lock -> assertMsg "Incorrect context provided" $ Set.member context lock.context
```

?

Reply:
Maybe chnage the whole function to
```
-- | Default implementation of `release` from the `Lockable` interface.
releaseImpl :
  ( HasCreate t
  , HasFromInterface t Lockable
  , HasToInterface t Lockable
  ) => Optional Lock -> (Optional Lock -> t) -> Release -> Update (ContractId Lockable)
releaseImpl currentLock setLock Release{context} = do
  let
    lock = fromSomeNote "Must be locked" currentLock
    newContext = context `Set.delete` lock.context
    updatedLock =
      if lock.lockType == Reentrant && not (Set.null newContext)
      then Some $ lock with context = newContext
      else None
  when (newContext == lock.context) $ abort "New context must be different from the current one"
  fmap toInterfaceContractId . create . setLock $ updatedLock
  ```
  I think it is easier to understand that way.

Reply:
I also think this is easier to understand. I just wonder about the error message "New context must be different from the current one" if the user provides an unrecognized context. It only describes why the function failed, not what the user did wrong / has to do to fix it. Something like "Can only unlock a known context" or similar is maybe more descriptive?

Reply:
Or as @matteolimberto-da suggested above: "Incorrect context provided"